rfcs/20180907-contrib-sunset.md

```diff
@@ -0,0 +1,282 @@
+| eager              | asimshankar      | move to core                         |
+| estimator          | ispirmustafa mikecase | move to tensorflow/estimator    |
+| factorization      | agarwal-ashish   | delete (rebuild in core pending)?    |
+| feature_column     | ispirmustafa     | move to core (experimental)          |
+| ffmpeg             | fredbertsch      | delete                               |
+| framework          |               | partially move to core, delete the rest |
+| fused_conv         |                  | delete                               |
+| gan                | joel-shor        | move to separate repo                |
+| gdr            |*byronyi (poxvoculi)  |                                      |
+| graph_editor       | purpledog        | delete                               |
+| grid_rnn           |(qlzh727 ebrevdo) | delete                               |
+| hadoop             | yongtang (mrry)  | move to tensorflow/io?               |
+| hooks              | ispirmustafa     | delete (already in core)             |
+| hvx                | satok16          | delete (redundant with NNAPI)        |
+| igfs (#22194)      | dmitrievanthony  | move to tensorflow/io                |
+| ignite (#22210)    | dmitrievanthony  | move to tensorflow/io                |
+| image              |                  | partial move to tensorflow/contrib?  |
+| input_pipeline     | rohan100jain     | delete                               |
+| integrate          | shoyer *mcoram   | move to tensorflow/scientific?       |
+| kafka              | yongtang (mrry)  | move to tensorflow/io?               |
+| keras              | fchollet         | delete (moved to tf.keras)           |
+| kernel_methods    | petrosmol rostami | move to tensorflow/estimator?        |
+| kfac               | duckworthd       | delete (moved to tensorflow/kfac)    |
+| kinesis            | yongtang (mrry)  | move to tensorflow/io?               |
+| labeled_tensor     | shoyer           | delete                               |
+| layers             |                  | partial move to tensorflow/contrib?  |
+| learn              | wicke       | delete (replaced by tensorflow/estimator) |
+| legacy_seq2seq     | ebrevdo (qlzh727) | delete (replaced by seq2seq)        |
+| libsvm             |                  | delete (no owner)                    |
+| linalg             | rmlarsen langmore | delete (moved to core)              |
+| linear_optimizer   | petrosmol (karmel) | move to tensorflow/estimator       |
+| lite             | aselle  petewarden | move to core                         |
+| lookup          | ysuematsu (ebrevdo) | move to core                         |
+| losses             |                  | partial move to tensorflow/contrib   |
+| makefile           |  petewarden      | delete (RPI build now uses bazel)    |
+| memory_stats       | wujingyue        | delete                               |
```

Comment:
Is there an alternative to query memory usage (especially on GPUs) ?

Reply:
You can ask `Session.run` to fill out [`StepStats`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/step_stats.proto), which includes information on memory usage.

Reply:
It does not support the functionality of the "MaxBytesInUseOp" in contrib, which returns me `AllocatorStats.max_bytes_in_use`. The closest thing in `StepStats` seems to be [AllocatorMemoryUsed.allocator_bytes_in_use](https://github.com/tensorflow/tensorflow/blob/9ab01c6732dae1143e22713375a9cc7758216787/tensorflow/core/framework/step_stats.proto#L32), which [corresponds to AllocatorStats.bytes_in_use](https://github.com/tensorflow/tensorflow/blob/d5c5f8ecc124ee9a866318f2bd7082df9e38ebf2/tensorflow/core/common_runtime/step_stats_collector.cc#L179-L181) instead.

Reply:
Is AllocatorMemoryUsed.peak_bytes equivalent to that?

Reply:
If I'm not mistaken, those are [__per-node__ statistics](https://github.com/tensorflow/tensorflow/blob/6419058dc6d065c34cccf55b11a7ba02c2ef7b06/tensorflow/core/framework/step_stats.proto#L22-L24). However the "MaxBytesInUse" op in contrib gives global peak memory usage.

Reply:
(Following the response in the main thread)
I tried and sum of `peak_bytes` for all node always return me a much larger results than the `MaxBytesInUse` op, sometimes close to 2x larger.
There is little docs on what these fields mean, but I guess sum of peak memory per node is expected to be much larger than the overall memory consumption (which I'm interested in) since nodes can have non-overlapped lifetime.
Conceptually, I think that the peak memory consumption of a whole system cannot be the sum of any per-node statistics.

Reply:
The (`AllocatorStats.max_bytes_in_use`)[https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/allocator.h#L54] should give you the peak memory per device. That op queries that field. You can also use another way to query that field: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/clusters/single_machine.cc#L237

Reply:
I think @ppwwyyxx is interested in querying this from Python. We don't have a way to do that any more. Should the peak device memory usage be added to StepStats (or something in there?)

Reply:
We can add a new field to StepStats to hold global peak memory per allocator if necessary.